gazebo/common/SdfFrameSemantics.cc

```diff
@@ -75,15 +83,18 @@ void convertPosesToSdf16(const sdf::ElementPtr &_modelElem)
   // SDF Model DOM object to be used for resolving poses with frame semantics
   sdf::Model modelSDFDom;
   sdf::Errors errors = modelSDFDom.Load(_modelElem);
-  if (!errors.empty())
+
+  for (const auto &error : errors)
   {
-    for (const auto &error : errors)
+    if (isSdfFrameSemanticsError(error))
     {
       gzerr << error << "\n";
     }
```

Comment:
is it safe to ignore the errors that are not FrameSemanticsErrors or they are handled in a different place?

Reply:
I see that it is a common pattern in more files under this PR.

Reply:
DOM objects are only used for frame semantics in Gazebo classic. We use the `sdf::Element` API for actually loading SDFormat worlds and models. So, IMO, errors reported by DOM objects are only relevant if they are related to frame semantics.